packages/email/email.js

```diff
@@ -145,17 +206,24 @@ Email.hookSend = function (f) {
  * `new EmailInternals.NpmModules.mailcomposer.module`.
  */
 Email.send = function (options) {
-  for (var i = 0; i < sendHooks.length; i++)
-    if (! sendHooks[i](options))
-      return;
-
   if (options.mailComposer) {
     options = options.mailComposer.mail;
   }
 
-  var transport = getTransport();
+  let send = true;
+  sendHooks.each(hook => {
+    send = hook(options);
+    return send;
+  });
+  if (!send) return;
+
+  const customTransport = Email.customTransport;
+  const transport = customTransport ? false : getTransport();
```

Comment:
too confusing, as the custom transport has priority we should first try to use it, the code will be much simpler.

also, you don't need else's here.

Reply:
So something like this:
```
const customTransport = Email.customTransport;
  if (customTransport) {
    const packageSettings = Meteor.settings.packages?.email || {};
    customTransport({ packageSettings: { ...packageSettings }, ...options, });
  } else if (Meteor.isProduction) {
    const transport = getTransport();
    smtpSend(transport, options);
  } else {
    devModeSend(options);
  }
```

Reply:
Almost, you don't need to have this variable `customTransport` and also you never need to spread something that is the full content of an object.

```js
{ packageSettings: { ...packageSettings }
```

should be

```js
{ packageSettings,
```

Reply:
Fixed